src/main/java/ch/njol/skript/expressions/ExprDifference.java

```diff
@@ -156,34 +151,27 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 
 	@Override
 	@Nullable
-	@SuppressWarnings({"unchecked", "rawtypes"})
+	@SuppressWarnings("unchecked")
 	protected Object[] get(Event event) {
 		Object first = this.first.getSingle(event);
 		Object second = this.second.getSingle(event);
 		if (first == null || second == null) {
 			return new Object[0];
 		}
 
-		Arithmetic math = this.math;
-		Class<?> relativeType = this.relativeType;
-
-		if (relativeType == Object.class) { // Try to determine now that actual types are known
-			ClassInfo<?> info = Classes.getSuperClassInfo(Utils.getSuperType(first.getClass(), second.getClass()));
-			math = info.getMath();
-			if (math == null) { // User did something stupid, just return <none> for them
+		if (differenceInfo == null) { // Try to determine now that actual types are known
+			Class<?> superType = Utils.getSuperType(first.getClass(), second.getClass());
+			differenceInfo = Arithmetics.getDifferenceInfo(superType);
```

Comment:
```suggestion
		DifferenceInfo differenceInfo = this.differenceInfo;
		if (differenceInfo == null) { // Try to determine now that actual types are known
			Class<?> superType = Utils.getSuperType(first.getClass(), second.getClass());
			differenceInfo = Arithmetics.getDifferenceInfo(superType);
```
We do not want to modify the actual field at runtime